File: goblinframework-core/src/main/java/org/goblinframework/core/transcoder/TranscoderFactoryImpl.java
package org.goblinframework.core.transcoder;

import org.goblinframework.api.core.Install;
import org.goblinframework.api.core.Ordered;
import org.jetbrains.annotations.NotNull;

@Install
@Deprecated
final public class TranscoderFactoryImpl implements TranscoderFactory, Ordered {

  @Override
  public int getOrder() {
    return LOWEST_PRECEDENCE;
  }

  @NotNull
  @Override
  public Transcoder1 buildTranscoder(@NotNull TranscoderSetting setting) {
    return new Transcoder1Impl(setting);
  }
}

File: goblinframework-transport/src/main/java/org/goblinframework/transport/core/codec/TranscoderFactoryImpl.kt
package org.goblinframework.transport.core.codec

import org.goblinframework.api.annotation.Install
import org.goblinframework.api.annotation.Singleton
import org.goblinframework.core.transcoder.Transcoder
import org.goblinframework.core.transcoder.TranscoderFactory
import org.goblinframework.core.transcoder.TranscoderSetting

@Singleton
class TranscoderFactoryImpl private constructor() : TranscoderFactory {

  companion object {
    @JvmField val INSTANCE = TranscoderFactoryImpl()
  }

  override fun buildTranscoder(setting: TranscoderSetting): Transcoder {
    return TranscoderImpl(setting)
  }

  @Install
  class Installer : TranscoderFactory by INSTANCE
}
